### DISCUSSION TITLE
Type casting doesn't work

### DISCUSSION THREAD

**Discussion by FelixLgr** [trivago/prettier-plugin-sort-imports#157] [Q&A]
Hello, I have seen this problem (https://github.com/trivago/prettier-plugin-sort-imports/issues/109 & https://github.com/trivago/prettier-plugin-sort-imports/issues/81).
In my case, it does not work. Does anyone have any idea why?
```js
const hasActiveType = filters.some((filter) => (<string[]>Object.values(InvoicesType)).includes(filter.value))
```
It return :
```json
  "code": "BABEL_PARSER_SYNTAX_ERROR",
  "reasonCode": "UnexpectedToken",
```

The problem come from `<string[]>`.

My config :
```json
  "importOrder": ["^src/(.*)$", "^[./]"],
  "importOrderSeparation": true,
  "importOrderSortSpecifiers": true,
  "importOrderGroupNamespaceSpecifiers": true,
  "importOrderParserPlugins": ["classProperties", "decorators-legacy", "typescript", "jsx"]
  ```

**Comment by steve-taylor** [comment]
Babel doesn't support angle-bracket type casting, so you might as well do this instead:

```ts
const hasActiveType = filters.some((filter) => (Object.values(InvoicesType) as string[]).includes(filter.value))
```

If you need any more reasons to use `as` to cast:

> TypeScript 1.6 introduces a new `.tsx` file extension. This extension does two things: it enables JSX inside of TypeScript files, and it makes the new `as` operator the default way to cast (removing any ambiguity between JSX expressions and the TypeScript prefix cast operator). For example:
>
> ```ts
> var x = <any>foo;
> // is equivalent to:
> var x = foo as any;
> ```

See https://www.typescriptlang.org/docs/handbook/release-notes/typescript-1-6.html


You might also have luck with

```ts
const hasActiveType = filters.some((filter) => (Object.values<string>(InvoicesType)).includes(filter.value))
```

**Comment by maou-shonen** [comment]
In one of the projects I manage, I also encountered this issue and came across this discussion.
I mostly agree with the point made by @steve-taylor.
Therefore, I added [consistent-type-assertions](https://typescript-eslint.io/rules/consistent-type-assertions/) to the ESLint rules,
hoping it can help save some time for others who encounter this issue in the future :)

```js
// .eslintrc
{
  rules: {
    '@typescript-eslint/consistent-type-assertions': [
      'error',
      {
        assertionStyle: 'as',
      },
    ],
  }
}
```
